shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Generator/Utxo.hs

```diff
@@ -136,174 +133,85 @@ genTx
-          --
-          -- Once the transaction body and the witnesses are constructed, we can use
-          -- this model to calculate the real fee value and update the fees and
-          -- transaction outputs in the final, generated transaction.
-          txBody <- genTxBody (Set.fromList inputs) outputs certs wdrls update (Coin 0) slotWithTTL
-          let multiSig =
-                Map.fromList $
-                  ( map
-                      ( \(payScript, _) ->
-                          (hashScript payScript, payScript)
-                      )
-                      spendScripts
-                  )
-                    ++ ( map
-                           ( \(_, sScript) ->
-                               (hashScript sScript, sScript)
-                           )
-                           (stakeScripts ++ wdrlScripts)
-                       )
-
-          -- choose one possible combination of keys for multi-sig scripts
-          --
-          -- TODO mgudemann due to problems with time-outs, we select one combination
-          -- deterministically for each script. Varying the script is possible though.
-
-          let wits =
-                mkTxWits
-                  ksIndexedPaymentKeys
-                  ksIndexedStakingKeys
-                  (hashAnnotated txBody)
-                  ( spendWitnesses
-                      ++ wdrlWitnesses
-                      ++ certWitnesses
-                      ++ updateWitnesses
-                      ++ genesisWitnesses
-                      ++ genesisDelegationWitnesses
-                  )
-                  multiSig
-
-          let metadata = SNothing -- TODO generate metadata
-
-          -- calculate real fees of witnesses transaction
-          let minimalFees = minfeeBound pparams (Tx txBody wits metadata)
-
-          -- discard generated transaction if the balance cannot cover the fees
-          if minimalFees > balance_
-            then QC.discard
-            else do
-              -- update model transaction with real fees and outputs
-              let (fees', outputs') =
-                    calcOutputsFromBalance balance_ recipientAddrs minimalFees
-                  txBody' =
-                    txBody
-                      { _txfee = fees',
-                        _outputs = outputs'
-                      }
-                  wits' =
-                    mkTxWits
-                      ksIndexedPaymentKeys
-                      ksIndexedStakingKeys
-                      (hashAnnotated txBody')
-                      ( spendWitnesses
-                          ++ wdrlWitnesses
-                          ++ certWitnesses
-                          ++ updateWitnesses
-                          ++ genesisWitnesses
-                          ++ genesisDelegationWitnesses
-                      )
-                      multiSig
-
-              pure (Tx txBody' wits' metadata)
+      let metadata = SNothing -- TODO generate metadata
```

Comment:
We just discovered [a bug](https://github.com/input-output-hk/cardano-node/issues/1545#issuecomment-663576625) that could have been found earlier if this TODO had been implemented :slightly_smiling_face:. Unrelated to this PR, but it should be fairly simple to add.

Reply:
I'm sorry to hear that @mrBliss :( Metadata is very coming soon, but clearly too late for this round